File: backend/src/main/java/com/ll/hotel/domain/booking/payment/dto/TokenResponse.java
package com.ll.hotel.domain.booking.payment.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true) // 불필요한 필드 무시
public record TokenResponse(
        int code,
        String message,
        ResponseData response
) {

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ResponseData(
            @JsonProperty("access_token") String accessToken,
            long now,
            @JsonProperty("expired_at") long expiredAt
    ) {}
}

File: backend/src/main/java/com/ll/hotel/domain/booking/payment/dto/TokenResponse.kt
package com.ll.hotel.domain.booking.payment.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true) // 불필요한 필드 무시
data class TokenResponse(
    val code: Int,
    val message: String,
    val response: ResponseData
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ResponseData(
        @JsonProperty("access_token") val accessToken: String,
        val now: Long,
        @JsonProperty("expired_at") val expiredAt: Long
    )
}